blog/2015/10/teaching_extreme_ranges.html

```diff
@@ -0,0 +1,117 @@
+(although being able to `cd` to a directory and knowing what a working directory does help).
+
+This workshop was different for me in that I've never had to teach a group with this wide of a skill gap.
+One one end, we had novices completely new to everything we had to offer in the workshop.
+On the other, we had a SysAdmin looking to transition Pearl scripts to Python.
+From the workshop-pre questionaire, everyone reported some language to the
+"With which programming languages, if any, could you write a program from scratch which imports some data and calculates mean and standard deviation of that data?"
+question.
+The least frequent coding frequency was "once a month".
+I went into the workshop almost afraid that I'll run out of things to talk about because
+many of the basic programming concepts in our material would be easy to pick up for individuals who have programming experience.
+My plan going into the workshop was to run though the inflammation material for Python, cover through loops and functions,
+and show them a little about using asserts to unit test functions and ask them for topics they wanted to cover for day 2.
+I had a few topics already planned out for Day 2: Reading in files, string parsing, and an intro to the <code>os</code> and <code>sys</code> modules and hoped to have a smorgasburg of topics that I got from the students the previous day when I asked them for feedback.
+In hindsight (and from feedback), this is when range of skills became an issue.
+
+Up until the end of the workshop, I imagined the pace I was moving was realatively okay.
+It could be a little fast at times but I figured the skills reported in the pre-workshop questionaire, and the
+<a href='https://docs.google.com/document/d/1djwDkkaaZh2PcQtuehg9NeRznsdRynIdom3pB807-ec/edit?usp=sharing'>feedback</a> at the end of the first day suggested that more advanced topics can be covered.
+It wasn't until the end of the day when I collected feedback, that I realized that I had lost a few people from the class.
+Attrition from the second day was relatively the same as I expected
+(not many people will be able to completely block off 2 full days from his/her schedule), any by the end of the day I only got 3 bits of feedback on sticky notes, 2 of which commented that the Python session was too fast and was a "waste of time", and how the student was disappointed because they were looking forward to the section.
+
+During my reflection of the workshop the past few weeks, I've realized my failure as an instructor to promote an environment
+where more novice students could feel they can ask questions and slow down the class on questions they had.
+
+My call to the rest of the community is how do we as instructors deal with extreme ranges in the classroom.
+Many of the solutions may be similar to a <a href='http://software-carpentry.org/blog/2015/10/pulling-along-those-behind.html'> recent post</a> by Peter Steinbach and Cam Macdonnell.
```

Comment:
Please use `<a href="{{page.root}}/blog/2015/10/...">...</a>` (i.e., `{{page.root}}` instead of `http://software-carpentry.org`).